localflavor/au/models.py

```diff
@@ -52,3 +53,26 @@ def formfield(self, **kwargs):
         defaults = {'form_class': forms.AUPhoneNumberField}
         defaults.update(kwargs)
         return super(AUPhoneNumberField, self).formfield(**defaults)
+
+
+class AUBusinessNumberField(CharField):
+    """
+    A model field that checks that the value is a valid Australian Business
+    Number (ABN).
+
+    .. versionadded:: 1.3
+    """
+
+    description = _("Australian Business Number")
+
+    validators = [AUBusinessNumberFieldValidator()]
+
+    def __init__(self, *args, **kwargs):
+        # ABN is 11 digits long, but we allow to enter it using spaces
+        kwargs['max_length'] = 14
```

Comment:
I don't think you should store the value with spaces in the database. With the current design, '5300 4085 616' and '53004085616' would not be considered the same value even though it's the same ABN. This means a unique constraint on the field would miss some cases.

I hit a similar problem with spaces when I developed the `IBANField`. You can remove the spaces in   `to_python()` of the model field like this:
https://github.com/django/django-localflavor/blob/master/localflavor/generic/models.py#L52

Doing this is not without issues though. If a users enters '5300 4085 616', they will see  '53004085616' after they save the form. It is also possible to re-add the spaces in the value on the form in `prepare_value`. Here's an example:
https://github.com/django/django-localflavor/blob/master/localflavor/generic/forms.py#L97

Of course this also has a problem: if a user enters ''53004085616', they will see '5300 4085 616' when the form is saved.

In the end I'm not sure how an ABN is normally displayed so I'd don't know if adding spaces automatically like the IBANField is best approach. You're free to decide what makes sense for this. Let me know if you agree with saving only the digits in the database.